src/NuGet.Services.AzureSearch/BaseDocumentBuilder.cs

```diff
@@ -175,16 +179,21 @@ public BaseDocumentBuilder(IOptionsSnapshot<AzureSearchJobConfiguration> options
 
             if (leaf.IconFile != null)
             {
-                var provider = new FlatContainerPackagePathProvider(_options.Value.FlatContainerContainerName);
-                var iconPath = provider.GetIconPath(document.PackageId, normalizedVersion);
-                document.IconUrl = new Uri(_options.Value.ParseFlatContainerBaseUrl(), iconPath).AbsoluteUri;
+                SetIconUrlFromFlatContainer(document);
             }
             else
             {
                 document.IconUrl = leaf.IconUrl;
             }
         }
 
+        private void SetIconUrlFromFlatContainer(IBaseMetadataDocument document)
```

Comment:
Do we need this functionality for other V3 jobs, like catalog2registration? If so, maybe a helper class/function is better rather than a private method here?